[BUG] Compiler warning (promoted to error) in Assisted factory with type params

## Describe the Bug

Followup to #995 - I apologize for not highlighting this in the original issue. Once the type parameters are present on the method, unfortunately the compiled code still misses them in one place

## Environment Info
12.5-RC4

```java
 /**
   * Fabricates a new DefaultDbLinkRegisteredProducer.
   */
  @Override
  public <K, V> DefaultDbLinkRegisteredProducer<K, V> create(String topicName, Properties props, Serde<K> keySerde, Serde<V> valueSerde) {
      var bean = new DefaultDbLinkRegisteredProducer(linker, topicName, props, keySerde, valueSerde);
      return bean;
  }
```

The constructor also needs to be type-parameterized, either with explicit `<K, V>` or implicit diamond `<>`, otherwise a warning -> error results:

```
[WARNING] COMPILATION WARNING : 
[WARNING] /.../kafka/config/DefaultDbLinkRegisteredProducer$AssistFactory.java:[31,18] unchecked call to DefaultDbLinkRegisteredProducer(com.....kafka.config.KafkaDbLinkLinker,java.lang.String,java.util.Properties,org.apache.kafka.common.serialization.Serde<K>,org.apache.kafka.common.serialization.Serde<V>) as a member of the raw type com.paywholesail.components.kafka.config.DefaultDbLinkRegisteredProducer
[WARNING] /.../kafka/config/DefaultDbLinkRegisteredProducer$AssistFactory.java:[32,14] unchecked conversion
  required: com....kafka.config.DefaultDbLinkRegisteredProducer<K,V>
  found:    com.....kafka.config.DefaultDbLinkRegisteredProducer
[INFO] 2 warnings 
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /.../DefaultDbLinkRegisteredProducer$AssistFactory.java: warnings found and -Werror specified
```

--- a/inject-generator/src/main/java/io/avaje/inject/generator/SimpleAssistWriter.java
+++ b/inject-generator/src/main/java/io/avaje/inject/generator/SimpleAssistWriter.java
@@ -248,7 +248,8 @@ final class SimpleAssistWriter {
   }
 
   private void writeCreateBean(MethodReader constructor) {
-    writer.start("var bean = new %s(", shortName);
+    var diamond = beanReader.beanType().getTypeParameters().isEmpty() ? "" : "<>";
+    writer.start("var bean = new %s%s(", shortName, diamond);
     // add constructor dependencies
     writeMethodParams(constructor, true);
   }

